src/ssh_client.py

```diff
@@ -0,0 +1,77 @@
+import paramiko
+import logging
+
+from src.base_client import BaseShellClient
+
+
+class SSHClient(BaseShellClient):
+    def __enter__(self):
+        try:
+            self.client = paramiko.SSHClient()
+            self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            self.client.connect(self.host, port=self.port, username=self.username, password=self.password)
+            logging.info(f"Connected to {self.host} successfully.")
+        except Exception as e:
+            logging.error(f"Failed to connect to {self.host}: {e}")
+            self.client = None
+        return self
+
+    def upload_file(self, local_path, remote_path):
+        if self.client is None:
+            raise ConnectionError("SSH connection not established.")
```

Comment:
It's enough to  raise ConnectionError("SSH connection not established.") only once ( in init or enter method). No needs to check if self.client is None in  each method